src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -71,6 +71,12 @@ public virtual void AddCompilationRoots(IRootingServiceProvider rootProvider)
 
             AddWellKnownTypes(rootProvider);
             AddReflectionInitializationCode(rootProvider);
+            foreach (var inputFile in _typeSystemContext.ReferenceFilePaths)
```

Comment:
This block should be placed in `SingleFileCompilationModuleGroup::AddCompilationRoots`. We don't want to generate exports for non-input modules when we're doing a multifile compilation (because those exports will be generated when we're compiling the referenced module itself).